### PR TITLE
Api contract roles

### PR DESCRIPTION
**Pull request by gokulakrishnansvm**
Added Api contract test case for roles endpoint
<hr/>

## Which Traffic Control components are affected by this PR?

- Traffic Ops Test

## What is the best way to verify this PR?

Step 1 : Install requirements
pip install -r requirements.txt

Step 2 : To execute roles API Contract Testcase , navigate to traffic_ops/testing/api_contract/v4
pytest --to-user Username --to-password Password --to-url URL --config Config Path --request-template Request template --response-template Response template test_roles.py

Note: Config, Request template and response template paths are optional. 

## PR submission checklist
- [x] This PR has tests 
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

File: traffic_ops/testing/api_contract/v4/conftest.py [hunk 1]
```diff
@@ -471,3 +471,52 @@ def cachegroup_post_data(to_session: TOSession, request_template_data: list[JSON
...
+
+
+@pytest.fixture()
+def role_post_data(to_session: TOSession, request_template_data: list[JSONData]
+			) -> dict[str, object]:
+	"""
+	PyTest Fixture to create POST data for roles endpoint.
+
+	:param to_session: Fixture to get Traffic Ops session.
+	:param api_prerequisite_data: Fixture to get role data from a prerequisites file.
```
Comment (ericholguin) [inline traffic_ops/testing/api_contract/v4/conftest.py:483]: should be `request_template_data` not `api_prerequisite_data`

File: traffic_ops/testing/api_contract/v4/test_roles.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+	request_template_data: list[dict[str, object] | list[object] | primitive],
+	response_template_data: dict[str, primitive | list[primitive | dict[str, object]
+						    | list[object]] | dict[object, object]],
+	role_post_data: dict[str, object]
+) -> None:
+	"""
+	Test step to validate keys, values and data types from roles endpoint
+	response.
+	:param to_session: Fixture to get Traffic Ops session.
+	:param request_template_data: Fixture to get role data from a prerequisites file.
```
Comment (ericholguin) [inline traffic_ops/testing/api_contract/v4/test_roles.py:39]: missing response_template_data

**Review by ericholguin — CHANGES_REQUESTED** [review]
minor changes, everything else looks good. Tests are passing.

File: traffic_ops/testing/api_contract/v4/conftest.py (hunk 1 above)
Comment (gokulakrishnansvm) [reply traffic_ops/testing/api_contract/v4/conftest.py]: Updated it!

File: traffic_ops/testing/api_contract/v4/test_roles.py (hunk 2 above)
Comment (gokulakrishnansvm) [reply traffic_ops/testing/api_contract/v4/test_roles.py]: Added it!

**Review by TerekhovaKate — APPROVED** [review]

**Comment by mitchell852** [comment]
@gokulakrishnansvm can you give this PR a better name. Something like "Added Api contract test case for roles endpoint"

**Pull request merged**